webview-ui/src/components/chat/hooks/useChatTextDraft.ts

```
@@ -0,0 +1,72 @@
+import { useCallback, useEffect, useRef } from "react"
+
+/**
+ * Hook for chat textarea draft persistence (localStorage).
+ * Handles auto-save, restore on mount, and clear on send.
+ * @param draftKey localStorage key for draft persistence
+ * @param inputValue current textarea value
+ * @param setInputValue setter for textarea value
+ * @param onSend send callback
+ */
+export function useChatTextDraft(
+	draftKey: string,
+	inputValue: string,
+	setInputValue: (value: string) => void,
+	onSend: () => void,
+) {
+	const saveDraftTimerRef = useRef<NodeJS.Timeout | null>(null)
+
+	// Restore draft on mount
+	useEffect(() => {
+		try {
+			const draft = localStorage.getItem(draftKey)
+			if (draft && !inputValue) {
+				setInputValue(draft)
+			}
+		} catch (_) {
+			// ignore
+		}
+		// Only run on initial mount
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [])
+
+	// Periodically save draft
+	useEffect(() => {
+		if (saveDraftTimerRef.current) {
+			clearInterval(saveDraftTimerRef.current)
+		}
+		if (inputValue && inputValue.trim()) {
+			saveDraftTimerRef.current = setInterval(() => {
+				try {
+					localStorage.setItem(draftKey, inputValue)
+				} catch (_) {
```

While silently catching errors prevents crashes, it might be helpful to log these failures for debugging. Could we add console.warn or use a logging utility to track when localStorage operations fail? This would help identify issues in environments where localStorage is restricted or full.